### ISSUE TITLE
Submitting to AppStore using IDFA fails

### ISSUE THREAD

**Issue by i0n1** [fastlane/fastlane#13055]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

When submitting to the app-store with IDFA it fails with:
/usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.101.1/spaceship/lib/spaceship/tunes/tunes_client.rb:1105:in `send_app_submission': [!] Something wrong with your Ad ID information: ["ITC.apps.idfa.errorMessage.agreeoptout", "ITC.apps.idfa.errorMessage.binaryviolatesterms"]. 

##### Complete output when running fastlane, including the stack trace and command used

<details>
I'm using the deliver command

This is the Deliverfile content:

creenshots_path "./screenshots"

metadata_path "./metadata"

app_rating_config_path "./metadata/rating.json"

price_tier 0

submit_for_review true

force true

automatic_release true

submission_information({
  export_compliance_available_on_french_store: "false",
  export_compliance_contains_proprietary_cryptography: "false",
  export_compliance_contains_third_party_cryptography: "false",
  export_compliance_is_exempt: "false",
  export_compliance_uses_encryption: "false",
  export_compliance_app_type: nil,
  export_compliance_encryption_updated: "false",
  export_compliance_compliance_required: "false",
  export_compliance_platform: "ios",
  content_rights_contains_third_party_content: "false",
  content_rights_has_rights: "false",
  add_id_info_limits_tracking: "true",
  add_id_info_serves_ads: "false",
  add_id_info_tracks_action: "true",
  add_id_info_tracks_install: "true",
  add_id_info_uses_idfa: "true"
})

  <pre>[07:46:40]: Successfully selected build
[07:46:40]: Submitting the app for review...
[07:46:44]: Setting 'export_compliance_available_on_french_store' to 'false'...
[07:46:44]: Setting 'export_compliance_contains_proprietary_cryptography' to 'false'...
[07:46:44]: Setting 'export_compliance_contains_third_party_cryptography' to 'false'...
[07:46:44]: Setting 'export_compliance_is_exempt' to 'false'...
[07:46:44]: Setting 'export_compliance_uses_encryption' to 'false'...
[07:46:44]: Setting 'export_compliance_app_type' to ''...
[07:46:44]: Setting 'export_compliance_encryption_updated' to 'false'...
[07:46:44]: Setting 'export_compliance_compliance_required' to 'false'...
[07:46:44]: Setting 'export_compliance_platform' to 'ios'...
[07:46:44]: Setting 'content_rights_contains_third_party_content' to 'false'...
[07:46:44]: Setting 'content_rights_has_rights' to 'false'...
[07:46:44]: Setting 'add_id_info_limits_tracking' to 'true'...
[07:46:44]: Setting 'add_id_info_serves_ads' to 'false'...
[07:46:44]: Setting 'add_id_info_tracks_action' to 'true'...
[07:46:44]: Setting 'add_id_info_tracks_install' to 'true'...
[07:46:44]: Setting 'add_id_info_uses_idfa' to 'true'...
Looking for related GitHub issues on fastlane/fastlane...
➡️  Deliver submission Ad ID information Error after 2.62.1 update
    https://github.com/fastlane/fastlane/issues/10735 [closed] 6 💬
    30 Dec 2017
➡️  [Deliver] Fix uses idfa limits tracking
    https://github.com/fastlane/fastlane/pull/10715 [closed] 16 💬
    16 Jan 2018
➡️  [Deliver] Deprecates add_id_info_limits_tracking in favor of add_id_info_uses_idfa
    https://github.com/fastlane/fastlane/pull/10653 [closed] 32 💬
    31 Dec 2017
🔗  You can ⌘ + double-click on links to open them directly in your browser.
/usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.101.1/spaceship/lib/spaceship/tunes/tunes_client.rb:1105:in `send_app_submission': [!] Something wrong with your Ad ID information: ["ITC.apps.idfa.errorMessage.agreeoptout", "ITC.apps.idfa.errorMessage.binaryviolatesterms"]. (RuntimeError)
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.101.1/spaceship/lib/spaceship/tunes/app_submission.rb:145:in `complete!'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.101.1/deliver/lib/deliver/submit_for_review.rb:27:in `submit!'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.101.1/deliver/lib/deliver/runner.rb:172:in `submit_for_review'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.101.1/deliver/lib/deliver/runner.rb:50:in `run'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.101.1/deliver/lib/deliver/commands_generator.rb:79:in `block (2 levels) in run'
	from /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	from /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.101.1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:74:in `run!'
	from /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.101.1/deliver/lib/deliver/commands_generator.rb:182:in `run'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.101.1/deliver/lib/deliver/commands_generator.rb:17:in `start'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.101.1/fastlane/lib/fastlane/cli_tools_distributor.rb:100:in `take_off'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.101.1/bin/fastlane:20:in `<top (required)>'
	from /usr/local/bin/fastlane:23:in `load'
	from /usr/local/bin/fastlane:23:in `<main>'</pre>
</details>

### Environment

<details>
  <pre>### Stack
| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.12.6                                     |
| Ruby                        | 2.4.2                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.15.1                          |
| Installation Source         | /usr/local/bin/fastlane                     |
| Host                        | Mac OS X 10.12.6 (16G1212)                  |
| Ruby Lib Dir                | /usr/local/Cellar/ruby/2.4.2_1/lib          |
| OpenSSL Version             | OpenSSL 1.0.2l  25 May 2017                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.2                                         |
### System Locale
| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |
### fastlane files:
**No Fastfile found**
**No Appfile found**
### fastlane gems
| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.101.1 | ✅ Up-To-Date  |
### Loaded fastlane plugins:
**No plugins Loaded**
<details><summary><b>Loaded gems</b></summary>
| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.1.0        |
| slack-notifier            | 2.3.2        |
| atomos                    | 0.1.3        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.5.9        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| terminal-notifier         | 1.8.0        |
| terminal-table            | 1.4.5        |
| plist                     | 3.1.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.2        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.6.3        |
| tty-cursor                | 0.6.0        |
| tty-spinner               | 0.8.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| excon                     | 0.59.0       |
| faraday                   | 0.13.1       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.1.3        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.12.2       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| dotenv                    | 2.2.1        |
| bundler                   | 1.16.0       |
| faraday_middleware        | 0.12.2       |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.5        |
| uber                      | 0.1.0        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.1        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| jwt                       | 2.1.0        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.8.1        |
| googleauth                | 0.6.2        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.23.4       |
| emoji_regex               | 0.1.1        |
| json                      | 2.0.4        |
| openssl                   | 2.0.5        |
| mini_portile2             | 2.1.0        |
| nokogiri                  | 1.6.8.1      |
</details>
*generated on:* **2018-08-07**</pre>
</details>

**Comment by Salierik** [comment]
UP!
I have the same issue :(

Fastlane version - 2.102.0

**Comment by DevZhaoXin** [comment]
UP!
I have the same issue :(

Fastlane version - 2.104.0

**Comment by frankspringtech** [comment]
same here

**Comment by wenjuan82** [comment]
+1 I have same issue 
Fastlane version - 2.104.0

**Comment by oliverchristen** [comment]
Using Fastlane version 2.107.0 and still experiencing this issue:
`Something wrong with your Ad ID information: ["ITC.apps.idfa.errorMessage.binaryviolatesterms"]`

However I do not see that error message: `ITC.apps.idfa.errorMessage.agreeoptout`

**Comment by Salierik** [comment]
Is there some features for request developer's attention or up priority?.. The ticket from 7 aug...

**Comment by Salierik** [comment]
UP

**Comment by ConfusedVorlon** [comment]
I'm seeing this with fastlane 2.113.0

```
submission_information({
  add_id_info_uses_idfa: true,
  add_id_info_serves_ads:true,
  add_id_info_tracks_action: false,
  add_id_info_tracks_install: false,
  add_id_info_limits_tracking: true,
  content_rights_contains_third_party_content: false,
  content_rights_has_rights: true,
  export_compliance_uses_encryption: false,
  export_compliance_encryption_updated: false
})
```


```
[14:16:21]: Successfully selected build
[14:16:21]: Submitting the app for review...
[14:16:30]: Setting 'add_id_info_uses_idfa' to 'true'...
[14:16:30]: Setting 'add_id_info_serves_ads' to 'true'...
[14:16:30]: Setting 'add_id_info_tracks_action' to 'false'...
[14:16:30]: Setting 'add_id_info_tracks_install' to 'false'...
[14:16:30]: Setting 'add_id_info_limits_tracking' to 'true'...
[14:16:30]: Setting 'content_rights_contains_third_party_content' to 'false'...
[14:16:30]: Setting 'content_rights_has_rights' to 'true'...
[14:16:30]: Setting 'export_compliance_uses_encryption' to 'false'...
[14:16:30]: Setting 'export_compliance_encryption_updated' to 'false'...
+--------------------+------------------------------------------------------------------------------------+
|                                              Lane Context                                               |
+--------------------+------------------------------------------------------------------------------------+
| DEFAULT_PLATFORM   | ios                                                                                |
| ENVIRONMENT        | free                                                                               |
| PLATFORM_NAME      | ios                                                                                |
| LANE_NAME          | ios submit                                                                         |
| BUILD_NUMBER       | 152                                                                                |
| IPA_OUTPUT_PATH    | /Users/rob/tmp/FastlaneBuild/VLCStreamerFree.ipa                                   |
| DSYM_OUTPUT_PATH   | /Users/rob/tmp/FastlaneBuild/VLCStreamerFree.app.dSYM.zip                          |
| XCODEBUILD_ARCHIVE | /Users/rob/Library/Developer/Xcode/Archives/2019-01-09/VLCStreamerFree 2019-01-09  |
|                    | 13.54.07.xcarchive                                                                 |
+--------------------+------------------------------------------------------------------------------------+
[14:16:36]: Something wrong with your Ad ID information: ["ITC.apps.idfa.errorMessage.agreeoptout"].
```

**Comment by joshdholtz** [comment]
Heyyyy 👋 I am not familiar with IDFA but I will gladly look into if I can. How do I go about submitting an app with this? 😇

**Comment by ConfusedVorlon** [comment]
Hi Josh,

not sure what info you're after. Docs on deliver are here
https://docs.fastlane.tools/actions/deliver/

I'm using 

```
  desc "Submit for review"
  lane :submit do
    increment_build_number
    gym
    deliver(
      force: true,
      skip_binary_upload: false,
      skip_screenshots: true,
      skip_metadata: false,
      submit_for_review: true,
      automatic_release: false)

  end
```

the submission information is in my deliverfile

**Comment by joshdholtz** [comment]
`When submitting to the app-store with IDFA it fails`

I don't know what "IDFA" means 😝

**Comment by joshdholtz** [comment]
Ahhhh, got it! So I have never used that before but I can create a sample app that does and then see if I can reproduce this error 💪🤞

**Comment by ConfusedVorlon** [comment]
excellent. 

Not sure what level of testing fastlane has - but it would be great if you do submission tests, it would be great to add a test with idfa enabled. This issue seems to come up every year or so.

any app with advertising hits this, so it is a pretty common case.

**Comment by orta** [comment]
This just hit me, so what we're seeing is this part of the submission process:

<img width="1414" alt="screen shot 2019-01-11 at 2 30 46 pm" src="https://user-images.githubusercontent.com/49038/51055294-7f40f600-15ad-11e9-96f4-52a8b7667f94.png">

Likely this new limit ad tracking section. That correlates to this property: `submitForReviewAnswers.adIdInfo.limitsTracking.value` [in the UI](https://user-images.githubusercontent.com/49038/51055185-22454000-15ad-11e9-90ca-60c75d86b3b6.png). 

There already is code to automatically [adds this](https://github.com/fastlane/fastlane/blob/master/spaceship/lib/spaceship/tunes/app_submission.rb#L144-L150) (which is why I'm commenting here and not making a PR, because it doesn't seem too obvious from a cursory glance yet)

**Comment by orta** [comment]
I fixed this with this commit: https://github.com/artsy/eigen/pull/2756/commits/25c9ddc956b88de77f3490bfa05dfccb01078d0c

Using @hjanuschka's example in https://github.com/fastlane/fastlane/issues/5542#issuecomment-254201994

**Comment by ConfusedVorlon** [comment]
@orta thanks for looking into this. Can you expound on what this means in terms of what is required in the submission_information for a successful submission?

**Comment by oliverchristen** [comment]
Fixed for me in v2.115.0
